### PR TITLE
followup #18711 grammar_nanny now reports unused non-terminals

### PR DESCRIPTION
**Pull request by timotheecour**
followup #18711 cleanup unused grammar rules

refs https://github.com/nim-lang/Nim/pull/18711#discussion_r692374660

### CODE REVIEW

File: tools/grammar_nanny.nim [hunk 1]
```diff
@@ -4,7 +4,7 @@
 import std / [strutils, sets]
 
 import ".." / compiler / [
-  llstream, ast, lexer, options, msgs, idents,
+  llstream, lexer, options, msgs, idents,
```
Comment (Araq) [inline tools/grammar_nanny.nim:7]: Why not patch grammar_nanny to automate this check?

Comment (timotheecour) [reply tools/grammar_nanny.nim:7]: done; it reports these:
```
Unused non-terminal: enum
Unused non-terminal: inlTupleDecl
Unused non-terminal: object
Unused non-terminal: procExpr
Unused non-terminal: exceptBlock
Unused non-terminal: tupleClass
Unused non-terminal: distinct
Unused non-terminal: tupleConstr
Unused non-terminal: module
Unused non-terminal: extTupleDecl
```
which can be addressed in future PR

Comment (Araq) [reply tools/grammar_nanny.nim:7]: Yes but now I need to look into why these are not used and what is used instead.

**Comment by Araq** [comment]
Simply removing these rules is not the proper fix, instead for example

```
primarySuffix = '(' (exprColonEqExpr comma?)* ')'
      | '.' optInd symbol generalizedLit?
...
```

should be patched to:

```
primarySuffix = '(' (exprColonEqExpr comma?)* ')'
      | '.' optInd symbol (generalizedLit? / '[:' exprList ']')
...
```

or whatever the parser actually implements...

**Comment by timotheecour** [comment]
PTAL, this PR now simple fixes the spurious import and adds the check in grammar nanny to report unused non-terminals, which is a useful thing to have; future work can fix those rules.

**Pull request merged**